lib/src/components/_internal_components.dart

```diff
@@ -127,33 +142,89 @@ class TimeLine extends StatelessWidget {
     required this.height,
     required this.timeLineOffset,
     required this.timeLineBuilder,
+    required this.liveTimeIndicatorSettings,
     this.showHalfHours = false,
   }) : super(key: key);
 
+  @override
+  State<TimeLine> createState() => _TimeLineState();
+}
+
+class _TimeLineState extends State<TimeLine> {
+  late Timer _timer;
+  late TimeOfDay _currentTime;
+  late DateTime _date;
+  late double _halfHourHeight;
+
+  @override
+  void initState() {
+    super.initState();
+    _date = DateTime.now();
+    _halfHourHeight = widget.hourHeight / 2;
+    _currentTime = widget.liveTimeIndicatorSettings.showTime
+        ? TimeOfDay.now()
+        : TimeOfDay(hour: 0, minute: 16);
+    _timer = Timer(Duration(seconds: 1),
+        widget.liveTimeIndicatorSettings.showTime ? setTimer : () {});
+  }
+
+  @override
+  void dispose() {
+    _timer.cancel();
+    super.dispose();
+  }
+
+  /// Creates an recursive call that runs every 1 seconds.
+  /// This will rebuild TimeLine every second. This will allow us
+  /// to show/hide time line when there is overlap with
+  /// live time line indicator in Week and Day view.
+  void setTimer() {
+    if (mounted) {
+      setState(() {
```

Comment:
Can we improve this to set the state only if the hour and minutes are actually changed?

Reply:
Also, use `Timer.periodic` if possible.

Reply:
@PRBaraiya Thanks for your feedback. Will try to incorporate those and update once done.